controllers/flowlogspipeline/flp_objects.go

```diff
@@ -508,6 +539,48 @@ func (b *builder) service(old *corev1.Service) *corev1.Service {
 	return newService
 }
 
+func (b *builder) promService(old *corev1.Service) *corev1.Service {
```

Comment:
I already didn't like much this pattern of "if service doesn't exist do ... else do ..." (I blame myself for that), now it spreads in the code, it's ok for this PR but at some point I'd like to see if we can have something more concise | readable | elegant ...

Reply:
I agree, but as your comment state this come from the immutable cluster ip field in the service.

I am curious to know how the kubectl cli works when updating a service. Does it first check existing service and enrich the new one? This would be strange since it add a lot of service specific code to kubectl.

Reply:
i'm not sure either